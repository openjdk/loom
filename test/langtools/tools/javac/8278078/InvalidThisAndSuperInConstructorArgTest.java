
import java.util.function.Supplier;

/**
 * @test /nodynamiccopyright/
 * @bug 8278078
 * @summary error: cannot reference super before supertype constructor has been called
 * @compile/fail/ref=InvalidThisAndSuperInConstructorArgTest.out -XDrawDiagnostics InvalidThisAndSuperInConstructorArgTest.java
 */
public class InvalidThisAndSuperInConstructorArgTest  {

    interface InterfaceWithDefault {
        default String get() {
            return "";
        }
    }

    InvalidThisAndSuperInConstructorArgTest(String s) {
    }

    class InnerClass extends AssertionError implements InterfaceWithDefault {
        InnerClass() {
            super(InnerClass.super.toString());
        }
        InnerClass(int i) {
            this(InnerClass.super.toString());
        }
        InnerClass(boolean b) {
            super(InnerClass.this.toString());
        }
        InnerClass(double d) {
            this(InnerClass.this.toString());
        }
        InnerClass(float f) {
            super(AssertionError.super.toString());
        }
        InnerClass(char ch) {
            this(AssertionError.super.toString());
        }
        InnerClass(byte b) {
            super(AssertionError.this.toString());
        }
        InnerClass(Object o) {
            this(AssertionError.this.toString());
        }
        InnerClass(int[] ii) {
            this(InterfaceWithDefault.super.get());
        }
        InnerClass(boolean[] bb) {
            super(InterfaceWithDefault.this.get());
        }
        InnerClass(double[] dd) {
            this(InterfaceWithDefault.this.get());
        }
        InnerClass(float[] ff) {
            super(InterfaceWithDefault.super.get());
        }
        InnerClass(char[] chch) {
            this(InnerClass.this::toString);
        }
        InnerClass(String s) {
            super(s);
        }
        InnerClass(Supplier<String> sup) {
            super(sup);
        }
    }
}
