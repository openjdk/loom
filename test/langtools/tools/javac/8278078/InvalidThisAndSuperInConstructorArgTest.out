InvalidThisAndSuperInConstructorArgTest.java:23:29: compiler.err.cant.ref.before.ctor.called: super
InvalidThisAndSuperInConstructorArgTest.java:26:28: compiler.err.cant.ref.before.ctor.called: super
InvalidThisAndSuperInConstructorArgTest.java:29:29: compiler.err.cant.ref.before.ctor.called: this
InvalidThisAndSuperInConstructorArgTest.java:32:28: compiler.err.cant.ref.before.ctor.called: this
InvalidThisAndSuperInConstructorArgTest.java:35:33: compiler.err.not.encl.class: java.lang.AssertionError
InvalidThisAndSuperInConstructorArgTest.java:38:32: compiler.err.not.encl.class: java.lang.AssertionError
InvalidThisAndSuperInConstructorArgTest.java:41:33: compiler.err.not.encl.class: java.lang.AssertionError
InvalidThisAndSuperInConstructorArgTest.java:44:32: compiler.err.not.encl.class: java.lang.AssertionError
InvalidThisAndSuperInConstructorArgTest.java:47:38: compiler.err.cant.ref.before.ctor.called: super
InvalidThisAndSuperInConstructorArgTest.java:50:39: compiler.err.not.encl.class: InvalidThisAndSuperInConstructorArgTest.InterfaceWithDefault
InvalidThisAndSuperInConstructorArgTest.java:53:38: compiler.err.not.encl.class: InvalidThisAndSuperInConstructorArgTest.InterfaceWithDefault
InvalidThisAndSuperInConstructorArgTest.java:56:39: compiler.err.cant.ref.before.ctor.called: super
InvalidThisAndSuperInConstructorArgTest.java:59:28: compiler.err.cant.ref.before.ctor.called: this
13 errors
